### PR TITLE
Added distinctMultiaddr method

### PR DESCRIPTION
**Pull request by SidHarder**
This method can be called from libp2p-swarm/lib/transport.js just before the createListeners method is called and may be a solution to issue https://github.com/ipfs/js-ipfs/issues/228.  This method returns an array of multiaddr which are distinct by port.  Transport is not considered, if there are two transports with the same port the first multiaddr will be returned.

Question:
1. Does this seem like a resonable solution?
2. Should both port and transport be used to determine what is distinct?

### CODE REVIEW

**Comment by diasdavid** [comment]
Almost! :) Unfortunately (or fortunately, I still don't know if there is a valid reason), TCP and UDP ports are different and some kernels also offer SCTP, which I'm not 100% sure if they share the same port space as UDP, that being said, making it unique by port would filter out valid addresses.

You also might want to use multiple network interfaces with the same port

**Comment by SidHarder** [comment]
I added to the transport object in [js-multiaddr](https://github.com/multiformats/js-multiaddr/pull/28) so that I could filter on it here. 

If you have this:
"Addresses": {
    "Swarm": [
      "/ip4/0.0.0.0/tcp/4001",
      "/ip6/::/tcp/4001",
      "/ip4/0.0.0.0/udp/4001",
      "/ip6/::/udp/4001"
    ],
    "API": "/ip4/127.0.0.1/tcp/5001",
    "Gateway": "/ip4/127.0.0.1/tcp/8080"
  },

The filter will return this:
"/ip4/0.0.0.0/tcp/4001"
"/ip4/0.0.0.0/udp/4001"

**Comment by diasdavid** [comment]
Looking good! Can you also add tests?

**Comment by SidHarder** [comment]
Yes, I will work on adding some tests.

**Comment by SidHarder** [comment]
diasdavid: There are two test that are failing for the repo now which I don't believe has anything to do with what I have been working on.  After looking at the 2 tests I don't see how the test would ever pass.  The 2 test are:

1) add multiaddr that are buffers
2) peer-info add multiaddr that are buffers:

Do you have any insight on this?

File: src/index.js [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 const Id = require('peer-id')
 const multiaddr = require('multiaddr')
+const _ = require('lodash')
```
Comment (diasdavid) [inline src/index.js]: use `const uniqBy = require('lodash.uniqBy')` if that is the only feature you are going to use, it saves in bundle size :)

Comment (SidHarder) [inline src/index.js]: got it!

**Comment by diasdavid** [comment]
CI looks fine to me, with the exception that SauceLabs is being wonky. where do you see those tests failing specifically @SidHarder ?

File: src/index.js (hunk 1 above)
Comment (SidHarder) [inline src/index.js]: David, I have not seen a require like this: require('lodash.uniqBy'),  did you mean like this: require('lodash').uniqBy? Or am I missing something?

**Pull request merged**